addons/delivery/models/delivery_carrier.py

```diff
@@ -131,6 +131,15 @@ def onchange_states(self):
     def onchange_countries(self):
         self.state_ids = [(6, 0, self.state_ids.filtered(lambda state: state.id in self.country_ids.mapped('state_ids').ids).ids)]
 
+    @api.model
+    def _neutralize(self):
+        super()._neutralize()
+        carriers = self.search([('active', '=', True)])
+        carriers.update({'active': False})
```

Comment:
Please avoid using method `update`, as it was designed mostly for onchange methods.
Use `write` or a direct assignment instead.